tests/unit/test_environment.py

```diff
@@ -8,109 +7,106 @@
+    """Tests location is saved correctly in the environment obj.
 
     Parameters
     ----------
-    example_env : rocketpy.Environment
-        Example environment object to be tested.
+    example_plain_env : rocketpy.Environment
+    latitude: float
+        The latitude in decimal degrees.
+    longitude: float
+        The longitude in decimal degrees.
     """
-    example_env.set_location(-21.960641, -47.482122)
-    assert example_env.latitude == -21.960641
-    assert example_env.longitude == -47.482122
+    example_plain_env.set_location(latitude, longitude)
+    assert example_plain_env.latitude == latitude
+    assert example_plain_env.longitude == longitude
 
 
-def test_set_elevation(example_env):
-    """Tests if the elevation is set correctly in the environment object. This
-    one basically set the elevation using the set_elevation() method and then
-    checks if the elevation is the same as the one in the example environment
-    object.
+@pytest.mark.parametrize("elevation", [(-200), (0), (200)])
+def test_elevation_set_elevation_saves_elevation(elevation, example_plain_env):
+    """Tests elevation is set correctly in the environment obj.
 
     Parameters
     ----------
-    example_env : rocketpy.Environment
-        Example environment object to be tested.
+    example_plain_env : rocketpy.Environment
     """
-    example_env.set_elevation(elevation=200)
-    assert example_env.elevation == 200
+    example_plain_env.set_elevation(elevation=elevation)
+    assert example_plain_env.elevation == elevation
 
 
-def test_set_topographic_profile(example_env):
-    """Tests the topographic profile in the environment object.
+@pytest.mark.parametrize(
+    "latitude, longitude, theoretical_elevation", [(46.90479, 8.07575, 1565)]
+)
+def test_location_set_topographic_profile_computes_elevation(
+    latitude, longitude, theoretical_elevation, example_plain_env
+):
+    """Tests elevation computation given topographic profile in the environment obj.
 
     Parameters
     ----------
-    example_env : rocketpy.Environment
-        Example environment object to be tested.
+    example_plain_env : rocketpy.Environment
+    latitude: float
+        The latitude in decimal degrees.
+    longitude: float
+        The longitude in decimal degrees.
     """
-    example_env.set_location(46.90479, 8.07575)
-    example_env.set_topographic_profile(
+    example_plain_env.set_topographic_profile(
         type="NASADEM_HGT",
         file="data/sites/switzerland/NASADEM_NC_n46e008.nc",
         dictionary="netCDF4",
     )
-    assert (
-        example_env.get_elevation_from_topographic_profile(
-            example_env.latitude, example_env.longitude
-        )
-        == 1565
+    computed_elevation = example_plain_env.get_elevation_from_topographic_profile(
+        latitude, longitude
     )
+    assert computed_elevation == theoretical_elevation
```

Comment:
This seems to be too restricted. What if we use pytest.approx to make this comparison?
It doesn't need to have a large tolerance, it is just to avoid numerical errors like: 5 is not equals to 5.0000000000001

Reply:
Thank you for your comment.
I was under the impression that the output of this method should be a bi-linearly interpolated value based on that array, that's why I thought about the float value problem.

Let's leave the code as it is right now, we revisit this if we ever face a floating problem.